sdk/eventhub/azure-eventhub/azure/eventhub/_client_base.py

```diff
@@ -174,7 +174,10 @@ def __init__(self, fully_qualified_namespace, eventhub_name, credential, **kwarg
         path = "/" + eventhub_name if eventhub_name else ""
         self._address = _Address(hostname=fully_qualified_namespace, path=path)
         self._container_id = CONTAINER_PREFIX + str(uuid.uuid4())[:8]
-        self._credential = credential
+        if isinstance(credential, AzureSasCredential):
+            self._credential =  EventHubSASTokenCredential(*parse_sas_credential(credential))
+        else:
+            self._credential = credential
```

Comment:
what if a user updates the signature in the `azure.core.credentials.AzureSasCredential`, I think our current `EventHubSASTokenCredential` implementation could not reflect the change?

we may need to either update the implementation of the `EventHubSASTokenCredential` of have a new private class, something like this:

```python
class XXXCredential:
    def __init__(self, core_sas_credential):
        self._core_sas_credetnail = core_sas_credential
    def get_token(self, *scopes, **kwargs):
        signature, expire = parse(core_sas_credential.signature)
        return AccessToken(signature, expiry)
```